### PR TITLE
Add tutorial for the transition from ROOT5 to ROOT6

### PR DESCRIPTION
**Pull request by mfasDa**
Basic howto as help for people needing to move from
ROOT5 to ROOT6, listing the main differences. Section
within the tutorial page yet to be specified.

### CODE REVIEW

**Comment by marcovanleeuwen** [comment]
Who needs to approve this request? Dario?

**Comment by marcovanleeuwen** [comment]
On the content; I realised that there is a fourth (?) way, which is to #include the code that needs to be loaded. I am using this for a bunch of utility routines that I use in plotting macros. It's maybe not the most elegant, but it does work, and for simple code, I think it is OK to go that way; it's definitely the most convenient. I can propose an edit/update mentioning this.

File: SUMMARY.md [hunk 1]
```diff
@@ -7,4 +7,5 @@
   * [Step 6 -Run your code on GRID](analysis/grid.md)
   * [Bonus content](analysis/bonus.md)
   * [Some hints](analysis/hints.md)
+  * [Tansition from ROOT5 to ROOT6](analysis/ROOT5-to-6.md)
```
Comment (dberzano) [inline SUMMARY.md]: Tansition -> Transition

File: analysis/ROOT5-to-6.md [hunk 2]
```diff
@@ -0,0 +1,339 @@
+# Transition from ROOT5 to ROOT6
+
+*ROOT6* is similar to *ROOT5*: It comes with the same library of classes (base
+classes, histogram classes like TH1, ...) which share the same interface as under
+*ROOT5*. It provides a command line interface as *ROOT5*, and runs macros just
+like under *ROOT5*. Therefore it is possible to build and run the ALICE
+simulation, reconstruction and analysis framework under ROOT6. AliPhysics can be
+build against *ROOT6* in the following way:
```
Comment (dberzano) [inline analysis/ROOT5-to-6.md]: build -> built

File: analysis/ROOT5-to-6.md [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+which would thrown by a compiler leading in a break of the compilation are still
+tolerated by *ROOT5*. Furthermore the standard library is not fully supported as
+its handling has to be implemented into the compiler. Macro can however also be
+run in the compiled mode by ROOT5 (adding a "+" to the macro name). In this case
+ROOT calls an external compiler to compile the macro to a library which is loaded
+into ROOT. Once the macro runs in compiled mode it is expected to be fully
+compatible with the C++ standard and coding error free as the compiler will not
+tolerate compiler errors.
+
+In contrast to this *ROOT6* uses a just-in-time compiler called *CLING*. A
```
Comment (dberzano) [inline analysis/ROOT5-to-6.md]: CLING -> cling

File: analysis/ROOT5-to-6.md [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+just-in-time compiler translates each statement, no matter whether it is a single
+command or a complicated macro, into machine code as it would be a program and
+only afterwards runs this piece of code. In order to translate code into machine
+code the compiler must know how to translate every single line. Based on powerful
+compiler library this means for *ROOT6* that all macro must comply to 100% with
+the C++ language standard, and exceptions will not be tolerated. Furthermore this
+also means that all symbols and objects must be known at compile time. This has
+three side effects:
+
+- Loadig Functions with ```gROOT->LoadMacro(...);```: This function loads all
```
Comment (dberzano) [inline analysis/ROOT5-to-6.md]: three backticks to one for inline code highlight (same applies later)

File: analysis/ROOT5-to-6.md [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+Calling macros inside macros is a bit tricky as the result of a macro will be
+available only at runtime. There are however several workarounds which cover the 
+most common use cases:
+
+- Using ROOT's TMacro: 
+
+  TMacro is a wrapper class around the macro processing. It is constructed with
+  the macro path. The macro is run using TMacro::Exec. The result of
+  TMacro::Exec() is a number representing an address in memory where to
+  find the resulting object. This has to be casted into a pointer to an object of
```
Comment (dberzano) [inline analysis/ROOT5-to-6.md]: casted -> cast

**Comment by mfasDa** [comment]
Hi @dberzano,

Thanks for the check! I addressed your requests. Can you please have a look whether it is fine now (with my markdown preview the inline syntax highlighting looks a bit strange now).

Cheers

Markus

**Comment by mfasDa** [comment]
Now it seems to look OK.

**Comment by dberzano** [comment]
Let's wait for the PR build and merge it.

**Comment by mfasDa** [comment]
Shall I maybe squash the commits?

**Comment by dberzano** [comment]
I have done it while merging, thanks!

**Pull request merged**